docs/csharp/advanced-topics/performance/index.md

```diff
@@ -33,44 +32,46 @@ When you *assign* a variable, you change its *value*. When you *ref assign* a va
-| `ref`, `in` parameter       | calling method                |
-| `out` parameter             | current method                |
-| `class` field               | calling method                |
-| non-ref `struct` field      | current method                |
-| `ref` field of `ref struct` | calling method                |
+| Declaration                           | *ref safe context*            |
+|---------------------------------------|-------------------------------|
+| non-ref local                         | block where local is declared |
+| non-ref parameter                     | current method                |
+| `ref`, `ref readonly`, `in` parameter | calling method                |
+| `out` parameter                       | current method                |
+| `class` field                         | calling method                |
+| non-ref `struct` field                | current method                |
+| `ref` field of `ref struct`           | calling method                |
 
-A variable can be `ref` returned if its *ref safe to escape scope* is the calling method. If its *ref safe to escape scope* is the current method or a block, `ref` return is disallowed. The following snippet shows two examples. A member field can be accessed from the scope calling a method, so a class or struct field's *ref safe to escape scope* is the calling method. The *ref safe to escape scope* for a parameter with the `ref`, or `in` modifiers is the entire method. Both can be `ref` returned from a member method:
+A variable can be `ref` returned if its *ref safe context* is the calling method. If its *ref safe context* is the current method or a block, `ref` return is disallowed. The following snippet shows two examples. A member field can be accessed from the scope calling a method, so a class or struct field's *ref safe context* is the calling method. The *ref safe context* for a parameter with the `ref`, or `in` modifiers is the entire method. Both can be `ref` returned from a member method:
 
 :::code language="csharp" source="./snippets/ref-safety/EscapeScopes.cs" id="RefSafeToEscapeScopes":::
 
 > [!NOTE]
-> When the `in` modifier is applied to a parameter, that parameter can be returned by `ref readonly`, not `ref`.
+> When the `ref readonly` or `in` modifier is applied to a parameter, that parameter can be returned by `ref readonly`, not `ref`.
 
-The compiler ensures that a reference can't escape its *ref safe to escape scope*. You can use `ref` parameters, `ref return` and `ref` local variables safely because the compiler detects if you've accidentally written code where a `ref` expression could be accessed when its storage isn't valid.
+The compiler ensures that a reference can't escape its *ref safe context*. You can use `ref` parameters, `ref return` and `ref` local variables safely because the compiler detects if you've accidentally written code where a `ref` expression could be accessed when its storage isn't valid.
```

Comment:
```suggestion
The compiler ensures that a reference can't escape its *ref safe context*. You can use `ref` parameters, `ref return`, and `ref` local variables safely because the compiler detects if you've accidentally written code where a `ref` expression could be accessed when its storage isn't valid.
```